### PR TITLE
Add tensorflow-records and tensorflow-records-conduit packages.

### PR DESCRIPTION
**Pull request by awpr**
The tensorflow-records package implements encoding/decoding of the
format, and the tensorflow-records-conduit package provides wrappers and
utilities for use with Conduit.

Thanks to @fkm3 for finding a native CRC32C implementation and writing the Cabal and Stack config for these packages.

Intended to supersede #70.

### CODE REVIEW

**Comment by fkm3** [comment]
Jenkins test this please.

File: tensorflow-records/src/TensorFlow/CRC32C.hs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+-- This masking is the algorithm specified by TensorFlow's TFRecords format.
+crc32cMask :: Word32 -> Word32
+crc32cMask x = rotateR x 15 + maskDelta
+
+-- | Inverse of 'crc32cMask'.
+crc32cUnmask :: Word32 -> Word32
+crc32cUnmask x = rotateL (x - maskDelta) 15
+
+-- | Convenience function combining 'crc32c' and 'crc32cMask'.
+crc32cMasked :: BL.ByteString -> Word32
```
Comment (fkm3) [inline tensorflow-records/src/TensorFlow/CRC32C.hs]: the compile is failing, this should be `B.ByteString`

**Comment by awpr** [comment]
Fixed, I wasn't able to convince Stack to build this before, so I was relying on the CI tests to catch problems.  I've got it working in Cabal now, though.

**Comment by fkm3** [comment]
Jenkins test this please.

**Comment by fkm3** [comment]
Thanks!

**Pull request merged**